geniza/corpus/tests/test_corpus_views.py

```diff
@@ -319,28 +319,35 @@ def test_get_queryset(self, mock_solr_queryset):
             mock_sqs.order_by.assert_called_with("-scholarship_count_i")
 
     @pytest.mark.usefixtures("mock_solr_queryset")
-    def test_get_context_data(self, rf, mock_solr_queryset):
+    @patch("geniza.corpus.views.DocumentSearchView.get_queryset")
+    def test_get_context_data(self, mock_get_queryset, rf, mock_solr_queryset):
         with patch(
             "geniza.corpus.views.DocumentSolrQuerySet",
             new=mock_solr_queryset(
                 DocumentSolrQuerySet, extra_methods=["admin_search", "keyword_search"]
             ),
         ) as mock_queryset_cls:
+
+            mock_qs = mock_queryset_cls.return_value
+            mock_qs.count.return_value = 22
+            mock_qs.get_facets.return_value.facet_fields = {}
+            # mock_qs.__getitem__.return_value = docsearch_view.queryset
+
+            mock_get_queryset.return_value = mock_qs
+
             docsearch_view = DocumentSearchView(kwargs={})
+            docsearch_view.queryset = mock_qs
+            docsearch_view.object_list = mock_qs
             docsearch_view.request = rf.get("/documents/")
-            docsearch_view.queryset = mock_queryset_cls.return_value
-            docsearch_view.queryset.count.return_value = 22
-            docsearch_view.queryset.get_facets.return_value.facet_fields = {}
-            docsearch_view.queryset.__getitem__.return_value = docsearch_view.queryset
-            docsearch_view.object_list = docsearch_view.queryset
 
             context_data = docsearch_view.get_context_data()
-            assert context_data["total"] == 22
             assert (
                 context_data["highlighting"]
                 == docsearch_view.queryset.get_highlighting.return_value
             )
             assert context_data["page_obj"].start_index() == 0
+            # NOTE: test paginator isn't initialized properly from queryset count
```

Comment:
Is this a bug with the `get_context_data`/paginator logic, or is it just happening because the paginator's not picking up the mock solr queryset? I see in my code I had used a separate `total` context var (didn't know about `paginator.count`), which just uses the `queryset.count`, so it makes sense that would have worked.

Reply:
I think it's because we're not running the normal progression of view methods that would happen if the view was dispatched normally, so the paginator isn't initialized from our mock. I hoped that setting the return from `get_queryset` would help, but I think there's something else not being called — I looked at the generic view mixin documentation for a bit and it wasn't obvious, so I decided not to worry about it for this test.

I was just simplifying the context data to remove the redundant total, since we can get it from the paginator.

Reply:
Makes sense to me!